src/XrdCrypto/XrdCryptosslAux.cc

```diff
@@ -568,12 +542,12 @@ int XrdCryptosslX509ParseFile(FILE *fcer,
 #endif
                         EVP_PKEY_free(evpp);
                         if (rc == 1) {
-                           DEBUG("RSA key completed");
-                           // Update PKI in certificate
+                           // Update PKI in certificate; also tests if the key is complete
                            cert->SetPKI((XrdCryptoX509data)rsa);
-                           // Update status
-                           cert->PKI()->status = XrdCryptoRSA::kComplete;
-                           break;
+                           if (cert->PKI()->status == XrdCryptoRSA::kComplete) {
+                              DEBUG("RSA key completed");
+                              break;
+                           }
```

Comment:
> Well, I don't completely understand how this will avoid repeated calls but it does look reasonable.

Just for the sake of documenting the change for later, the check is removed from `if (XrdCheckRSA(rsa) == 1)` which becomes just `if (rsa)` because later on in `SetPKI(rsa)` the check is performed when calling `new XrdCryptosslRSA(rsa, 1)`. The constructor ensures that the key being created is checked.